### DISCUSSION TITLE
Bot can't connect to network? Connect to ZNC instead!

### DISCUSSION THREAD

**Discussion by ph818** [knrd1/chatgpt#33] [General]
Some networks require identification with methods that the chatgpt bot doesn't support, like SASL. Thankfully, znc is a very mature IRC bouncer that supports a wide range of connection options. 

You can lookup a guide for setting up znc on your distro, and then connect to it with the bot. For example, this is the [IRC] section of my chat.conf

```
[irc]
server = localhost
port = 61234
ssl = false
channels = #linux
nickname = zncuser
ident = whatever
realname = chatbot
password = yourpasswordhere
```

where zncuser and yourpasswordhere are the credentials you set up for connecting to znc.

just make sure you name your znc admin user the same thing as your bot's nickname, because the 'nickname' in chat.conf will be used to login to znc, and also to recognize when users are talking to the bot.

**Comment by knrd1** [comment]
Nice one, thank you!
